plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -87,34 +87,43 @@
     static final String INFORMATION_SCHEMA = "information_schema";
     private static final String VIEW_DEFINITION_SYSTEM_TABLE_SUFFIX = "$view_definition";
 
-    private final BigQueryClient bigQueryClient;
-    private final String projectId;
+    private final BigQueryClientFactory bigQueryClientFactory;
+    private final Optional<String> configProjectId;
 
     @Inject
-    public BigQueryMetadata(BigQueryClient bigQueryClient, BigQueryConfig config)
+    public BigQueryMetadata(BigQueryClientFactory bigQueryClientFactory, BigQueryConfig config)
     {
-        this.bigQueryClient = requireNonNull(bigQueryClient, "bigQueryClient is null");
-        this.projectId = requireNonNull(config, "config is null").getProjectId().orElse(bigQueryClient.getProjectId());
+        this.bigQueryClientFactory = requireNonNull(bigQueryClientFactory, "bigQueryClientFactory is null");
+        this.configProjectId = requireNonNull(config, "config is null").getProjectId();
+    }
+
+    protected String getProjectId(BigQueryClient client)
+    {
+        return configProjectId.orElse(client.getProjectId());
```

Comment:
Add a verify that it's always lowercase. I remember some code depends on this.

Reply:
BigQuery does the validation (https://cloud.google.com/resource-manager/docs/creating-managing-projects) but it's nice to make that assumption explicit.